src/GraphQL.DataLoader.Tests/Types/QueryType.cs

```diff
@@ -76,6 +76,21 @@ public QueryType(IDataLoaderContextAccessor accessor, IUsersStore users, IOrders
                     return ret;
                 });
 
+            Field<NonNullGraphType<ListGraphType<UserType>>, IDataLoaderResult<IEnumerable<User>>>()
+                .Name("SpecifiedUsersWithThen")
+                .Description("Get Users by ID skipping null matches")
+                .Argument<NonNullGraphType<ListGraphType<NonNullGraphType<IntGraphType>>>>("ids")
+                .Resolve(ctx =>
+                {
+                    var loader = accessor.Context.GetOrAddBatchLoader<int, User>("GetUserById",
+                        users.GetUsersByIdAsync);
+
+                    var ids = ctx.GetArgument<IEnumerable<int>>("ids");
+                    var ret = ids.Select(id => loader.LoadAsync(id)).ToList();
```

Comment:
```suggestion
                    // note: does not work properly without ToList, because LoadAsync would not have
                    // been called, so the ids would not have been queued for execution prior to the
                    // first call to GetResultAsync
                    var ret = ids.Select(id => loader.LoadAsync(id)).ToList();
```